### PR TITLE
Dispatcher defs & submitter decls for kt::gpu::radix_sort

### PR DESCRIPTION
**Pull request by joeatodd**
This is the 3rd PR for the pure SYCL onesweep radix_sort implementation.

This PR builds on top of (and depends on) #1243, adding dispatcher definitions and submitter declarations.

### CODE REVIEW

File: include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h [hunk 1]
```diff
@@ -20,16 +20,157 @@
 #include <cassert>
 
 #include "../kernel_param.h"
-
-#include "../../../pstl/hetero/dpcpp/parallel_backend_sycl_utils.h"
```
Comment (dmitriy-sobolev) [inline include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h:24]: It defines `__kernel_name_provider`. Lets keep it.

File: include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h [hunk 2]
```diff
@@ -20,16 +20,157 @@
...
-#include "../../../pstl/hetero/dpcpp/utils_ranges_sycl.h"
```
Comment (dmitriy-sobolev) [inline include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h:25]: It defines `oneapi::dpl::__ranges::__get_sycl_range`. Let's keep it.

File: include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h [hunk 3]
```diff
@@ -20,16 +20,157 @@
...
+        __gpu_radix_sort_onesweep_odd<_KernelName>>;
+    using _GpuRadixSortCopyback = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
+        __gpu_radix_sort_onesweep_copyback<_KernelName>>;
+
+    using _GlobalHistT = ::std::uint32_t;
+    constexpr ::std::uint32_t __bin_count = 1 << __radix_bits;
+
+    // TODO: consider adding a more versatile API, e.g. passing special kernel_config parameters for histogram computation
+    constexpr ::std::uint32_t __hist_work_group_count = 64;
+    constexpr ::std::uint32_t __hist_work_group_size = 64;
```
[the remark this answers is not in the document]
Comment (joeatodd) [reply include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h]: Yeah I think the `TODO` above this line serves as a reminder to profile this & potentially hook it up to the kernel template params.

File: include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_submitters.h [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+#include <cstdint>
+
+namespace oneapi::dpl::experimental::kt::gpu::__impl
+{
+
+//------------------------------------------------------------------------
+// Please see the comment for __parallel_for_submitter for optional kernel name explanation
+//------------------------------------------------------------------------
+
+template <typename _KeyT, ::std::uint32_t __radix_bits, ::std::uint32_t __stage_count,
```
[the remark this answers is not in the document]
Comment (joeatodd) [reply include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_submitters.h]: Done, thanks!

File: include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h [hunk 5]
```diff
@@ -20,16 +20,157 @@
...
+    const ::std::uint32_t __tmp_keys_mem_size = __n * sizeof(_KeyT);
+    const ::std::size_t __tmp_mem_size = __group_hists_mem_size + __global_hist_mem_size + __tmp_keys_mem_size;
 
+    ::std::uint8_t* __p_tmp_mem = sycl::malloc_device<::std::uint8_t>(__tmp_mem_size, __q);
+    // Memory to store global histograms, where each stage has its own histogram
+    _GlobalHistT* __p_global_hist_all = reinterpret_cast<_GlobalHistT*>(__p_tmp_mem);
+    // Memory to store group historgrams, which contain offsets relative to "previous" groups
+    _GlobalHistT* __p_group_hists_all = reinterpret_cast<_GlobalHistT*>(__p_tmp_mem + __global_hist_mem_size);
+    // Memory to store intermediate results of sorting
+    _KeyT* __p_keys_tmp = reinterpret_cast<_KeyT*>(__p_tmp_mem + __global_hist_mem_size + __group_hists_mem_size);
```
[the remark this answers is not in the document]
Comment (joeatodd) [reply include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h]: I copied over your `__onesweep_memory_holder`. Thanks :)

File: include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h (hunk 2 above)
Comment (joeatodd) [reply include/oneapi/dpl/experimental/kt/internal/gpu_radix_sort_dispatchers.h]: Done :+1:

**Comment by joeatodd** [comment]
@dmitriy-sobolev thanks a lot for taking the time to review, and for your fixes. I think this is ready - what do you think?  Cheers

**Review by dmitriy-sobolev — APPROVED** [review]

**Pull request merged**